### PR TITLE
[Notifier] Add FakeChat Logger transport

### PR DESCRIPTION
**Pull request by noniagriconomie**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | Sub part of #40625
| License       | MIT
| Doc PR        | WIP

Refs https://github.com/symfony/symfony/pull/42123 for Notifier FakeChat

### CODE REVIEW

File: src/Symfony/Component/Notifier/Bridge/FakeChat/Tests/FakeChatLoggerTransportTest.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        yield [new SmsMessage('0611223344', 'Hello!')];
+        yield [$this->createMock(MessageInterface::class)];
+    }
+
+    public function testSendWithDefaultTransport()
+    {
+        $message1 = new ChatMessage($subject1 = 'Hello subject1!', new TestOptions(['recipient_id' => $recipient = 'Oskar']));
+        $message2 = new ChatMessage($subject2 = 'Hello subject2!');
+
+        $logger = new TestLogger();
```
Comment (noniagriconomie) [inline src/Symfony/Component/Notifier/Bridge/FakeChat/Tests/FakeChatLoggerTransportTest.php:52]: Ref https://github.com/symfony/symfony/pull/42123#discussion_r670412095

**Review by OskarStark — APPROVED** [review]

File: src/Symfony/Component/Notifier/Bridge/FakeChat/Tests/FakeChatLoggerTransportTest.php [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+use Symfony\Component\Notifier\Transport\TransportInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+final class FakeChatLoggerTransportTest extends TransportTestCase
+{
+    public function createTransport(HttpClientInterface $client = null, LoggerInterface $logger = null): TransportInterface
+    {
+        $transport = new FakeChatLoggerTransport($logger ?? $this->createMock(LoggerInterface::class), $client ?? $this->createMock(HttpClientInterface::class));
+
+        return $transport;
```
Comment (OskarStark) [inline src/Symfony/Component/Notifier/Bridge/FakeChat/Tests/FakeChatLoggerTransportTest.php]:
```suggestion
        return new FakeChatLoggerTransport($logger ?? $this->createMock(LoggerInterface::class), $client ?? $this->createMock(HttpClientInterface::class));
```

**Comment by OskarStark** [comment]
Please extend `UnsupportedSchemeExceptionTest`, thanks

File: src/Symfony/Component/Notifier/Bridge/FakeChat/Tests/TestLogger.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\FakeChat\Tests;
+
+use Psr\Log\AbstractLogger;
+
+class TestLogger extends AbstractLogger
```
Comment (OskarStark) [inline src/Symfony/Component/Notifier/Bridge/FakeChat/Tests/TestLogger.php]: Final please

Comment (noniagriconomie) [reply src/Symfony/Component/Notifier/Bridge/FakeChat/Tests/TestLogger.php]: ok lets do this

> Please extend UnsupportedSchemeExceptionTest, thanks

I do not understand this comment @OskarStark :s

Comment (OskarStark) [reply src/Symfony/Component/Notifier/Bridge/FakeChat/Tests/TestLogger.php]: Sorry, please open this file and add a test case for your transport, thanks

Comment (OskarStark) [reply src/Symfony/Component/Notifier/Bridge/FakeChat/Tests/TestLogger.php]: Does it makes sense to you?

**Review by derrabus — CHANGES_REQUESTED** [review]
Can you have a look at the Psalm errors?

File: src/Symfony/Component/Notifier/Bridge/FakeChat/FakeChatTransportFactory.php [hunk 4]
```diff
@@ -38,19 +42,22 @@ public function create(Dsn $dsn): TransportInterface
     {
         $scheme = $dsn->getScheme();
 
-        if ('fakechat+email' !== $scheme) {
```
Comment (OskarStark) [inline src/Symfony/Component/Notifier/Bridge/FakeChat/FakeChatTransportFactory.php:41]: Please use `if` for consistency, I already created a PR for the `FakeSmsLoggerTransport`:
https://github.com/symfony/symfony/pull/42340

Thanks

**Review by OskarStark — CHANGES_REQUESTED** [review]
see comments

File: src/Symfony/Component/Notifier/Bridge/FakeChat/FakeChatTransportFactory.php (hunk 4 above)
Comment (noniagriconomie) [reply src/Symfony/Component/Notifier/Bridge/FakeChat/FakeChatTransportFactory.php:41]: oki lets do this

Comment (OskarStark) [reply src/Symfony/Component/Notifier/Bridge/FakeChat/FakeChatTransportFactory.php:41]: Only psalm to fix

**Comment by noniagriconomie** [comment]
friendly ping @OskarStark @derrabus code reworked as requested, thx

**Review by OskarStark — APPROVED** [review]

**Review by derrabus — APPROVED** [review]

**Comment by OskarStark** [comment]
Thanks Antoine.

**Pull request merged**